include/deal.II/grid/grid_generator.h

```diff
@@ -1577,6 +1577,12 @@ namespace GridGenerator
    * @note For a related operation on refined meshes when both meshes are
    * derived from the same coarse mesh, see
    * GridGenerator::create_union_triangulation().
+   *
+   * @note By internally calling Triangulation::create_create_triangulation(),
+   * this function triggers the "create" signal after doing its work. See
```

Comment:
```suggestion
   * @note This function triggers the "create" signal after doing its work. See
```
We don't need to document what we do internally which might be subject to change.

Would you mind adding that it is legal for one of the input triangulations to be equal to `result` in response to https://github.com/dealii/dealii/issues/11622?